tests/js/server/utils-tests.js

```diff
@@ -0,0 +1,126 @@
+/*
+
+  Tests for the gpii.express.user.utils component.
+
+ */
+/* eslint-env node */
+"use strict";
+
+var fluid        = require("infusion");
+var gpii         = fluid.registerNamespace("gpii");
+
+require("../../../");
+require("../lib/");
+
+var jqUnit       = require("node-jqunit");
+
+fluid.registerNamespace("gpii.tests.express.user.utils.caseHolder");
+
+gpii.tests.express.user.utils.createUser = function (utils) {
+    var prom = utils.createNewUser({
+        username: "myFirstUser",
+        password: "this is a password",
+        email: "myFirstUser@gpii.net"
+    });
+    prom.then(function (data) {
+        jqUnit.assertEquals("Generated Couch ID", "org.couch.db.user:myFirstUser", data._id);
+        jqUnit.assertEquals("Email", "myFirstUser@gpii.net", data.email);
+        jqUnit.assertEquals("Username", "myFirstUser", data.username);
+        jqUnit.assertTrue("Unlock the password", gpii.express.user.utils.verifyPassword(data, "this is a password"));
```

Comment:
OK, I can see that this second check probably means you need a function rather than directly calling jqUnit.deepEq from your test definition.

Reply:
I'm not sure what this means...I'm not using deepEq any where.  Or if it's the assertEquals I'm not sure which structure change your looking for.

Reply:
My point was,  you *could* use jqUnit.deepEq for the first three comparisons, and for other function-based checks in your tests.  However, in this case you perform a secondary check that requires inspecting the results twice, which is IMO a good rule-of-thumb for making a function for the test vs. writing it declaratively.